front/app/components/InitiativeCards/index.tsx

```diff
@@ -198,17 +200,47 @@ interface Props {
   invisibleTitleMessage: MessageDescriptor;
 }
 
-const InitiativeCards = ({ className, invisibleTitleMessage }: Props) => {
-  const defaultFilters = {
-    publication_status: 'published' as InitiativePublicationStatus,
-  };
+interface QueryParameters {
+  publication_status: 'published';
+  sort: Sort;
+  search?: string;
+  initiative_status?: string;
+  topics?: string[];
+}
 
+const InitiativeCards = ({ className, invisibleTitleMessage }: Props) => {
   const { formatMessage } = useIntl();
   const { windowWidth } = useWindowSize();
-  const [selectedView, setSelectedView] = useState<'map' | 'card'>('card');
+
+  const [searchParams] = useSearchParams();
+
+  const sortParam = searchParams.get('sort') as Sort | null;
+  const searchParam = searchParams.get('search');
+  const initiativeStatusParam = searchParams.get('initiative_status');
+  const topicsParam = searchParams.get('topics');
+  const selectedInitiativeMarkerId = searchParams.get('initiative_map_id');
+
+  const selectedView =
+    (searchParams.get('view') as 'card' | 'map' | null) ??
+    (selectedInitiativeMarkerId ? 'map' : 'card');
+
+  const setSelectedView = useCallback((view: 'card' | 'map') => {
+    updateSearchParams({ view });
+  }, []);
+
   const [filtersModalOpened, setFiltersModalOpened] = useState(false);
-  const [selectedInitiativeFilters, setSelectedInitiativeFilters] =
-    useState<Partial<IQueryParameters>>(defaultFilters);
+
+  const selectedInitiativeFilters = useMemo<QueryParameters>(
```

Comment:
Why memoize these? Doesn't seem necessary.

Reply:
It was a `useState` before, so a stable reference. `useMemo` keeps it a stable reference. might not be necessary but if there's any `useEffect` downstream somewhere relying on this we're creating an infinite loop in our code if we don't use `useMemo`. Why take the risk?

Reply:
It's a conversation about the tradeoff between introducing complexity now for arguably no gain vs later and only in case we need to. Even though I see your point, I would still go with the first option. However, I don't consider it a blocking issue.